addons/payment_xendit/tests/test_payment_transaction.py

```diff
@@ -12,6 +12,30 @@
 @tagged('post_install', '-at_install')
 class TestPaymentTransaction(PaymentHttpCommon, XenditCommon):
 
+    def test_no_item_missing_from_rendering_values(self):
+        """ Test that the when redirect flow is triggered, rendering_values contain the
+        API_URL corresponding to the response of API request. """
+        tx = self._create_transaction(flow='redirect')
+        with patch(
+            'odoo.addons.payment_xendit.models.payment_provider.PaymentProvider'
+            '._xendit_make_request', return_value={'invoice_url': 'https://dummy.com'}
+        ):
+            rendering_values = tx._get_specific_rendering_values(None)
+        self.assertDictEqual(rendering_values, {'api_url': 'https://dummy.com'})
+
+    def test_empty_rendering_values_if_direct(self):
+        """ Test that if it's a card payment (like in direct flow), rendering_values should be empty
+        and no API call should be committed in the process"""
+        card_pm = self.env.ref('payment.payment_method_card').id
+        tx = self._create_transaction(flow='direct', payment_method_id=card_pm)
+        with patch(
+            'odoo.addons.payment_xendit.models.payment_provider.PaymentProvider'
+            '._xendit_make_request', return_value={'data': {'link': 'https://dummy.com'}}
+        ) as mock:
+            rendering_values = tx._get_specific_rendering_values(None)
+            mock.assert_not_called()
```

Comment:
What does the error message look like when this fails? For tests calling `mock.assert_called_once` when had to do `assertEqual(mock.call_count, 1)` instead because the error message was unhelpful.

Reply:
I think the error message is also unhelpful. Going to use `assertEqual(mock.call_count, 0)` in this case. Thanks!